### PR TITLE
Draw pach DAGs on the command line

### PR DESCRIPTION
**Pull request by acohen4**
TODO:
- [x] Think through test assertions
- [x] Implement an ordering algorithm
- [x] Add non-planar test case

### CODE REVIEW

File: src/server/pps/pretty/draw.go [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+	"github.com/fatih/color"
+	"github.com/google/uuid"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+)
+
+const (
+	// maxLabelLen        = 10 // TODO: How do we handle long pipeline names?
+	boxWidth           = 11
+	layerVerticalSpace = 5
```
Comment (acohen4) [inline src/server/pps/pretty/draw.go:16]: should we parameterize for CLI?

File: src/server/pps/server/api_server.go [hunk 2]
```diff
@@ -2407,16 +2397,43 @@ func (a *apiServer) listPipeline(ctx context.Context, request *pps.ListPipelineR
 		}
 		return true
 	}
-
-	for i := range infos {
-		if filterPipeline(infos[i]) {
-			if err := a.getLatestJobState(ctx, infos[i]); err != nil {
+	if err := ppsutil.ListPipelineInfo(ctx, a.pipelines, request.Pipeline, request.History, func(ptr *pps.PipelineInfo) error {
+		pi := proto.Clone(ptr).(*pps.PipelineInfo)
+		if request.CommitSet != nil && request.CommitSet.ID != "" {
```
Comment (albscui) [inline src/server/pps/server/api_server.go:2402]: try request.GetCommitSet() instead

**Review by albscui — CHANGES_REQUESTED** [review]
Maybe split this into 2 PRs, where 1st PR does the PPS changes, and the 2nd is purely the draw stuff.

File: src/server/pps/pretty/draw.go [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+	// calculate offset based on distance from the end in case we're on the bottom half of the edge
+	if vertIdx > vertDist/2 {
+		offset = adjustedXDist - (vertDist - offset)
+	}
+	if re.src > re.dest {
+		i := re.src - offset
+		return setEdgeChar(row, i, '/')
+	} else {
+		i := re.src + offset
+		return setEdgeChar(row, i, '\\')
```
Comment (albscui) [inline src/server/pps/pretty/draw.go:330]: Similarly
```suggestion
		return setEdgeChar(row, re.src + offset, '\\')
```
I prefer to un-nest this else and just return below the `if` block.

**Review by albscui — APPROVED** [review]
Overall LGTM, just a few nits on style.

**Pull request merged**